### PR TITLE
Use ruff instead of flake8, autoflake and isort

### PR DESCRIPTION
**Pull request by Kludex**
Ref.: https://github.com/encode/httpx/discussions/2607

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -19,19 +19,13 @@ build==0.10.0
...
-flake8==3.9.2
-flake8-bugbear==23.1.20
-flake8-pie==0.16.0; python_version>='3.7'
-importlib-metadata==4.13.0; python_version>='3.7'
-isort==5.11.4; python_version<'3.8'
-isort==5.12.0; python_version>='3.8'
 mypy==1.0.1
 types-certifi==2021.10.8.2
 pytest==7.2.2
+ruff==0.0.260
```
Comment (tomchristie) [inline requirements.txt:28]: This is nice

File: pyproject.toml [hunk 2]
```diff
@@ -90,3 +90,11 @@ text = "\n---\n\n[Full changelog](https://github.com/encode/httpx/blob/master/CH
 [[tool.hatch.metadata.hooks.fancy-pypi-readme.substitutions]]
 pattern = 'src="(docs/img/.*?)"'
 replacement = 'src="https://raw.githubusercontent.com/encode/httpx/master/\1"'
+
+[tool.ruff]
+select = ["E", "F", "I", "B", "PIE"]
+ignore = ["B904", "B028"]
+line-length = 120
```
Comment (tomchristie) [inline pyproject.toml:97]: Should we comment above here with a pointer to a relevant part of the docs?

File: tests/client/test_async_client.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ async def test_access_content_stream_response(server):
 
     assert response.status_code == 200
     with pytest.raises(httpx.ResponseNotRead):
-        response.content
+        response.content  # noqa: B018
```
Comment (tomchristie) [inline tests/client/test_async_client.py:87]: How can I track down what a B018 is?

Comment (Kludex) [reply tests/client/test_async_client.py:87]: It's on the rules page: https://beta.ruff.rs/docs/rules/#flake8-bugbear-b

Should I add a comment here or ignore it globally?

File: pyproject.toml (hunk 2 above)
Comment (Kludex) [reply pyproject.toml:98]: I've added a reference to the docs

Comment (Kludex) [reply pyproject.toml:98]: Is it enough?

Comment (tomchristie) [reply pyproject.toml:98]: 'tis

**Review by tomchristie — APPROVED** [review]

**Pull request merged**